glean-core/ffi/src/event.rs

```diff
@@ -0,0 +1,127 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::collections::HashMap;
+use std::convert::TryFrom;
+use std::iter::Iterator;
+use std::os::raw::c_char;
+
+use ffi_support::{define_handle_map_deleter, ConcurrentHandleMap, FfiStr};
+use lazy_static::lazy_static;
+
+use glean_core::metrics::{EventMetric, MetricType};
+use glean_core::{CommonMetricData, Lifetime};
+
+use crate::handlemap_ext::HandleMapExtension;
+use crate::{from_raw_string_array, RawIntArray, RawStringArray, GLEAN};
+
+lazy_static! {
+    pub static ref EVENT_METRICS: ConcurrentHandleMap<EventMetric> = ConcurrentHandleMap::new();
+}
+define_handle_map_deleter!(EVENT_METRICS, glean_destroy_event_metric);
+
+#[no_mangle]
+pub extern "C" fn glean_new_event_metric(
+    category: FfiStr,
+    name: FfiStr,
+    send_in_pings: RawStringArray,
+    send_in_pings_len: i32,
+    lifetime: i32,
+    disabled: u8,
+    extra_keys: RawStringArray,
+    extra_keys_len: i32,
+) -> u64 {
+    EVENT_METRICS.insert_with_log(|| {
+        let send_in_pings = unsafe { from_raw_string_array(send_in_pings, send_in_pings_len) };
```

Comment:
Uff, I really need to work on these unsafe parts to make them safer.

Reply:
@mdboom could you file a bug about this? Or @badboy is there any low-hanging fruit we could grab in this PR?

Reply:
I'm not clear on what could be done, but I filed [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1564523).

Reply:
Let's merge this as is and fix that in a followup (after https://github.com/mozilla/glean/pull/179 lands)